### PR TITLE
[OPPRO-327] Add map type validation

### PR DESCRIPTION
**Pull request by JkSelf**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by JkSelf** [comment]
@rui-mo @zhztheplayer Can you help to review? Thanks.

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala [hunk 1]
```diff
@@ -81,6 +79,7 @@ trait BasicScanExecTransformer extends TransformSupport {
       case byte: ByteType =>
       case array: ArrayType =>
       case bool: BooleanType =>
+      case bool: MapType =>
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:82]: Maybe `case map: MapType` is better here?

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala [hunk 2]
```diff
@@ -234,4 +234,23 @@ class VeloxDataTypeValidationSuite extends WholeStageTransformerSuite {
...
+
+    // Validation: BroadHashJoin, Filter, Project
+    super.sparkConf.set("spark.sql.autoBroadcastJoinThreshold", "10M")
+    runQueryAndCompare("select type1.map['key'] from type1," +
+      " type2 where type1.map['key'] = type2.map['key']") { _ => }
+
+    // Validation: ShuffledHashJoin, Filter, Project
+    super.sparkConf.set("spark.sql.autoBroadcastJoinThreshold", "-1")
+    runQueryAndCompare("select type1.map['key'] from type1," +
+      " type2 where type1.map['key'] = type2.map['key']") { _ => }
```
Comment (rui-mo) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala:254]: I see this PR added some support for map type in both cpp code and ArrowWritableColumnVector, will these code be executed in above unit test? Actually I'm not clear whether this test is executed by native or fallbacked into vanilla spark.

Comment (JkSelf) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala:254]: @rui-mo 
In this test, the Project, Limit and Filter Node will be offload to native and other operators will fallback to vanilla spark. So the ArrowWritableColumnVector code will be executed.

Here is one of the physical plan:
`
== Physical Plan ==

CollectLimit 1
+- *(1) ColumnarToRow
   +- VeloxLoadArrowData
      +- *(2) ProjectExecTransformer [map#14]
         +- VeloxRowToArrowColumnar
            +- BatchScan[map#14] ParquetScan DataFilters: [], Format: parquet, Location: InMemoryFileIndex(1 paths)[file:/home/sparkuser/jk/projects/gluten/backends-velox/target/scala-2...., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<map:map<string,string>>, PushedFilters: [] RuntimeFilters: []
`

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala (hunk 1 above)
Comment (JkSelf) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala]: Updated.

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala (hunk 2 above)
Comment (rui-mo) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala:254]: Thanks for the explain!

**Review by rui-mo — APPROVED** [review]

**Pull request merged**